MegRiley/cqf-ruler
# Meg riley oauth

Attempted to add OAuth following the same model that was used to configure the keycloak server

Included changes
- **fhirserver.properties** - A new file. This is a basic  prop file that contains, the realm name, client secret.
- **FhirAuthConfig.java** - A new file. This processes the contents of fhirserver.properties.
- **TransactionInterceptor.java** - This is an existing file. There were new methods added 
    buildRuleList(), applyRulesAndReturnDecision(), and makeDecision().  2 of the 3 are overrides, a 
   case could be made that perhaps they should live in a separate file but 
  **TransactionInterceptor.java** already includes  handleOutgoing/incomingPreprocessedHandled, 
     which is where the rules should be be checked (based on the behavior in the HAPI base classes.)

To Test:
- Start keycloak in standalone mode. (I set my instance up according to the instructions here https://github.com/HL7-DaVinci/CRD) 
- Run cqf-ruler 
- Using postman configured for bearer authentication and the  base STU3 url for your cqf-ruler
- You should be able to send a post message and get a 200 back (I'll be honest I'm not sure how to 
  be sure cqf-ruler was working.)

Note this is my first time working with cqf-ruler and rest so I'm sure I made some mistakes somewhere.
